tests/__init__.py

```diff
@@ -1,18 +1,62 @@
+import os
+import sys
+
+from diff_match_patch import diff_match_patch
+
 from drf_spectacular.validation import validate_schema
 
 
-def assert_schema(schema, reference_file):
+def assert_schema(schema, reference_filename, transforms=None):
     from drf_spectacular.renderers import OpenApiJsonRenderer, OpenApiYamlRenderer
 
     schema_yml = OpenApiYamlRenderer().render(schema, renderer_context={})
     # render also a json and provoke serialization issues
     OpenApiJsonRenderer().render(schema, renderer_context={})
 
-    with open(reference_file.replace('.yml', '_out.yml'), 'wb') as fh:
+    if not os.path.exists(reference_filename):
+        generated_filename = reference_filename
+    else:
+        generated_filename = reference_filename.replace('.yml', '_out.yml')
+
+    with open(generated_filename, 'wb') as fh:
         fh.write(schema_yml)
 
-    with open(reference_file) as fh:
-        assert schema_yml.decode() == fh.read()
+    if generated_filename == reference_filename:
+        raise RuntimeError(
+            f'{reference_filename} was not found. '
+            f'It has been created with the generated schema. '
+            f'Review carefully, as it is the baseline for subsequent checks.')
+
+    generated = schema_yml.decode()
+
+    with open(reference_filename) as fh:
+        expected = fh.read()
+
+    # This only does each transform, however it may be necessary to do all
+    # combinations of transforms.
+    # Force to be a list, and insert identity
+    if transforms:
+        transforms = [lambda x: x] + list(transforms)
+    else:
+        transforms = [lambda x: x]
+
+    results = []
+    for result in (transformer(generated) for transformer in transforms):
+        if result not in results:
+            results.append(result)
+            if result == expected:
+                break
+    else:
+        dmp = diff_match_patch()
+        diff = dmp.diff_main(expected, generated)
+        dmp.diff_cleanupSemantic(diff)
+        patch = dmp.patch_toText(dmp.patch_make(diff))
```

Comment:
This library is better than nothing, and quite good for short textual differences, especially within a single line, but the current results are far from perfect for large chunks of multiline differences.

The outstanding rest-polymorphic difference [here](https://travis-ci.org/github/tfranzel/drf-spectacular/jobs/709884465#L396) is not exactly easy to grok.

Maybe the invocation can be improved by tweaking some of its options.

Doing a plain unified diff is also an option; that at least leverages existing brain circuitry for people familiar with diffs of YAML.